test/ExecutionFramework.cpp

```diff
@@ -282,3 +282,21 @@ bool ExecutionFramework::storageEmpty(h160 const& _addr) const
 	}
 	return true;
 }
+
+vector<solidity::frontend::test::LogRecord> ExecutionFramework::recordedLogs() const
+{
+	vector<solidity::frontend::test::LogRecord> logs{};
+	for (size_t logIdx = 0; logIdx < m_evmcHost->recorded_logs.size(); ++logIdx)
+	{
+		solidity::frontend::test::LogRecord record;
+		const auto& data = m_evmcHost->recorded_logs.at(logIdx).data;
+		record.index = logIdx;
+		record.data = bytes{data.begin(), data.end()};
+		record.creator = EVMHost::convertFromEVMC(m_evmcHost->recorded_logs.at(logIdx).creator);
+		for (size_t topicIdx = 0; topicIdx < m_evmcHost->recorded_logs.at(logIdx).topics.size(); ++topicIdx) {
```

Comment:
Hmm, couldn't we just use a range loop:
```
for (auto const& topic: m_evmcHost->recorded_logs.at(logIdx).topics)
    record.topics.emplace_back(EVMHost::convertFromEVMC(topic));
```

You may not want to use auto, depending on who you ask :wink: